base/loading.jl

```diff
@@ -547,7 +547,7 @@ function stale_cachefile(modpath, cachefile)
             return true # invalid cache file
         end
         modules, files = cache_dependencies(io)
-        if files[1][1] != modpath
+        if realpath(files[1][1]) != realpath(modpath)
```

Comment:
I wonder if we should switch this to using `samefile` instead. Basically, I'm trying to think through how we want it to fail when one of these files (the symlink or the target) disappears.